plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergFileOperations.java

```diff
@@ -509,6 +510,59 @@ public void testSelectWithFilter()
                         .build());
     }
 
+    @Test
+    public void testPartialPartitionPruningEffectivenessWithPartitionTransform()
```

Comment:
nit: I would include "timestamp" to this test name. We will probably add similar tests with other data types.